Format negative datetimes with rfc3339 (#9501)

Taywee opened the pull request:
- fixes #9501 

# Description

`chrono::Datetime::to_rfc2822()` panics when it would format a negative year.  3339 does not.  This makes negative-year datetimes inconsistent compared to positive ones, but it's better than a panic.

amtoine commented on crates/nu-protocol/src/value/mod.rs:
```diff
@@ -12,7 +12,7 @@ use crate::engine::EngineState;
 use crate::{did_you_mean, BlockId, Config, Span, Spanned, Type, VarId};
 use byte_unit::ByteUnit;
-use chrono::{DateTime, Duration, FixedOffset};
+use chrono::{DateTime, Datelike, Duration, FixedOffset};
```
do we need `chrono::DateLike` to call `.to_rfc3339`?

Taywee replied:
No, but we do need it to call `.year()`.

amtoine replied:
ooooh i see, thanks for the explaination :+1:

amtoine reviewed (changes requested):
i was able to check this solves the issue :ok_hand: 

i've just got a question and also **could you add a test** that makes sure this issue is solved? :yum:

Taywee commented:
I'd be happy to add a test.  I'll do that right now.

Taywee commented:
I have added tests and validated that the negative year test fails in the previous state and succeeds with the patch.

I had to cut off the humanized stamp, because it's relative to the current time and therefore not test stable, but I don't feel too bad about it, for a small hack.

fdncred reviewed (approved):
Thanks! I'm good with these changes.

amtoine reviewed (approved):
thanks @Taywee, only a little merge resolution and that should be ok :yum:

Taywee commented:
Merge conflict resolved.  I rebased it for a cleaner history.

amtoine commented:
@Taywee 
no need to rebase inside a PR :wink: 
- it breaks the PR review process because the commits that have been reviewed are not part of the PR anymore
- we squash and rebase the PR onto the `main` branch anyway

Taywee commented:
My mistake.  To be completely honest, I also tried to fix the merge conflict in the GitHub editor and managed to break the PR in the process, so I also rebased to remove the ugly broken commit from the history, but if nu PRs are always rebased and squashed anyway, I suppose that didn't matter. :sweat_smile: 

I'll avoid unnecessary rebases in the future.  Thanks for the heads up.

fdncred commented:
Thanks!
